### DISCUSSION TITLE
problem opening current file in vim

### DISCUSSION THREAD

**Discussion by tkossak** [Taitava/obsidian-shellcommands#381] [Q&A] [answered]
Hi, I created command for `vim {{file_path:absolute}}` but when I run it nothing happens. I would expect the terminal would open with the file opened in vim. 
When I change command to `kwrite {{file_path:absolute}}` then kwrite is opened properly.
What am I missing? Why It doesn't run vim, not even terminal with any kind of error?

I tried:
- entering full vim path: /usr/bin/vim

I use:
- Linux

**Comment by Syakhisk** [comment, marked as answer]
vim is just a cli program, in order to see it you would need to run it inside a terminal. so try running it with your terminal.
For kitty terminal, I use:
```
/path/to/kitty nvim
```

If you use other terminal you might need to pass an arguments before the command, e.g.
```
/path/to/konsole -e nvim
```

**Reply by Taitava** [reply]
Hi! Thank you @tkossak for your question, and thank you @Syakhisk for your answer! 🙂 

> vim is just a cli program, in order to see it you would need to run it inside a terminal.

This is true. The _Shell commands_ plugin does not open up a terminal window (unless the executed command does), instead it executes shell commands as background processes, so programs requiring a terminal user interface will not work unless a terminal program is first invoked in the shell command itself.

In case your Linux is e.g. Ubuntu, you could try the following terminal paths:

- `/usr/bin/xterm vim {{file_path:absolute}}`, or
- `/usr/bin/gnome-terminal vim {{file_path:absolute}}`
- (These paths are found from https://askubuntu.com/a/753070)

> When I change command to `kwrite {{file_path:absolute}}` then kwrite is opened properly.

This is because [kwrite](https://apps.kde.org/kwrite/) is a graphical application, so it does not require a terminal. Those work directly. 🙂 

---

- [ ] I think I could add these instructions somewhere in the documentation.

Edit: I moved the discussion from _Debugging and testing_ to _Q & A_. 👍

**Comment by tkossak** [comment]
Thank you very much, `/usr/bin/konsole -e vim {{file_path:absolute}}` works on my Manjaro KDE perfectly :)
